src/rectangle_spawning/rectangle_entity.rs

```diff
@@ -0,0 +1,82 @@
+use bevy::prelude::*;
+use bevy::sprite::{MaterialMesh2dBundle, Mesh2dHandle};
+
+// Represents the dimensions of a rectangle
+#[derive(Debug, Clone, Copy, Component)]
+pub struct RectangleDimension(f32);
+
+impl RectangleDimension {
+    pub fn new(value: f32) -> Option<Self> {
+        if value > 0.0 {
+            Some(Self(value))
+        } else {
+            None
+        }
+    }
+
+    pub fn value(&self) -> f32 {
+        self.0
+    }
+}
+
+impl Default for RectangleDimension {
+    fn default() -> Self {
+        Self(1.0)
+    }
+}
+
+// Configuration for the rectangle's properties
+#[derive(Debug, Default, Clone, Component)]
```

Comment:
```suggestion
#[derive(Resource, Debug, Default, Clone)]
```

Should be a resource, since all tiles share the same configuration.